sw/device/silicon_creator/manuf/base/ft_personalize.c

```diff
@@ -587,6 +591,77 @@ static status_t personalize_gen_dice_certificates(ujson_t *uj) {
   return OK_STATUS();
 }
 
+static status_t compute_tbs_was_hmac(perso_blob_t *perso_blob_to_host) {
+  // Read out the WAS from flash.
+  hmac_key_t was;
+  static_assert(
+      kFlashInfoFieldWaferAuthSecretSizeIn32BitWords == kHmacKeyNumWords,
+      "WAS size expected to be same size as HMAC-SHA256 key.");
+  TRY(flash_ctrl_testutils_info_region_setup_properties(
+      &flash_ctrl_state, kFlashInfoFieldWaferAuthSecret.page,
+      kFlashInfoFieldWaferAuthSecret.bank,
+      kFlashInfoFieldWaferAuthSecret.partition, kFlashInfoPage3ReadPermissions,
+      /*offset=*/NULL));
+  TRY(manuf_flash_info_field_read(
+      &flash_ctrl_state, kFlashInfoFieldWaferAuthSecret, was.key,
+      kFlashInfoFieldWaferAuthSecretSizeIn32BitWords));
+
+  // Compute HMAC of TBS certs with WAS as the key.
+  // Most HSMs and host tooling will compute an HMAC in big endian format, so we
+  // do the same to make comparison easier.
```

Comment:
pedantic nitpicking: _All_ host tooling will compute the HMAC in big endian format, because there really is no endianness to SHA256 outputs.  It's really just a bag of bytes.

We think about endianness because our peripheral is word-oriented and has endianness configuration bits.

ECDSA and RSA keys, OTOH, can be thought of as having endianness because they key components are integers and any integer larger than a natural word size has to have an ordering in memory.

Reply:
reworded comment